test case failing when run with Python 3.7

In 2.4.0, I see one test case failing when run with Python 3.7. All tests succeed wihen run in 2.7.15 and 3.6.6.

running build_ext
test_checkers_testsuite (testsuite.test_all.PycodestyleTestCase) ... /home/packages/tmp/pycodestyle-2.4.0/testsuite/W29.py:13:1: W391:1:1 blank l
ine at end of file
/home/packages/tmp/pycodestyle-2.4.0/testsuite/W29.py:14:1: error W292:1:5 not found
/home/packages/tmp/pycodestyle-2.4.0/testsuite/W29.py:15:1: W391:1:1 blank line at end of file
/home/packages/tmp/pycodestyle-2.4.0/testsuite/W29.py:16:1: error W292:1:27 not found
/home/packages/tmp/pycodestyle-2.4.0/testsuite/W29.py:17:1: W391:1:1 blank line at end of file
/home/packages/tmp/pycodestyle-2.4.0/testsuite/W29.py:18:1: error W292:3:22 not found
/home/packages/tmp/pycodestyle-2.4.0/testsuite/W29.py:21:1: W391:3:1 blank line at end of file
FAIL
test_doctest (testsuite.test_all.PycodestyleTestCase) ... ok
test_own_dog_food (testsuite.test_all.PycodestyleTestCase) ... /home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:292:13: W504 line break after 
binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:350:13: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:356:21: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:357:21: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:378:13: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:379:13: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:380:13: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:610:30: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:656:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:657:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:664:15: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:741:13: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:742:13: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:743:13: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:745:13: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:747:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1112:14: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1113:14: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1239:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1240:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1272:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1273:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1274:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1308:24: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1739:38: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:1860:45: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:2212:17: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:2275:22: W504 line break after binary operator
/home/packages/tmp/pycodestyle-2.4.0/pycodestyle.py:2304:17: W504 line break after binary operator
ok
test_selftest (testsuite.test_all.PycodestyleTestCase) ... ok
test_check_nullbytes (testsuite.test_api.APITestCase) ... ok
test_check_unicode (testsuite.test_api.APITestCase) ... ok
test_register_ast_check (testsuite.test_api.APITestCase) ... ok
test_register_invalid_check (testsuite.test_api.APITestCase) ... ok
test_register_logical_check (testsuite.test_api.APITestCase) ... ok
test_register_physical_check (testsuite.test_api.APITestCase) ... ok
test_styleguide (testsuite.test_api.APITestCase) ... ok
test_styleguide_check_files (testsuite.test_api.APITestCase) ... ok
test_styleguide_checks (testsuite.test_api.APITestCase) ... ok
test_styleguide_continuation_line_outdented (testsuite.test_api.APITestCase) ... ok
test_styleguide_excluded (testsuite.test_api.APITestCase) ... ok
test_styleguide_ignore_code (testsuite.test_api.APITestCase) ... ok
test_styleguide_init_report (testsuite.test_api.APITestCase) ... ok
test_styleguide_options (testsuite.test_api.APITestCase) ... ok
test_styleguide_unmatched_triple_quotes (testsuite.test_api.APITestCase) ... ok
test_blank_line_between_decorator (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_blank_line_decorator (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_initial_lines_more_blank (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_initial_lines_one_blank (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_initial_lines_two_blanks (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_initial_no_blank (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_method_fewer_follow_lines (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_method_less_blank_lines (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_method_less_blank_lines_comment (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_method_more_blank_lines (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_method_nested_fewer_follow_lines (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_method_nested_less_class (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_method_nested_ok (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_top_level_fewer_blank_lines (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_top_level_fewer_follow_lines (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_top_level_fewer_follow_lines_comments (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_top_level_good_follow_lines (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_top_level_more_blank_lines (testsuite.test_blank_lines.TestBlankLinesDefault) ... ok
test_initial_lines_one_blanks (testsuite.test_blank_lines.TestBlankLinesTwisted) ... ok
test_initial_lines_tree_blanks (testsuite.test_blank_lines.TestBlankLinesTwisted) ... ok
test_the_right_blanks (testsuite.test_blank_lines.TestBlankLinesTwisted) ... ok
test_top_level_fewer_blank_lines (testsuite.test_blank_lines.TestBlankLinesTwisted) ... ok
test_top_level_more_blank_lines (testsuite.test_blank_lines.TestBlankLinesTwisted) ... ok
test_multiline_ignore_parsing (testsuite.test_parser.ParserTestCase) ... ok
test_multiline_trailing_comma_ignore_parsing (testsuite.test_parser.ParserTestCase) ... ok
test_trailing_comma_ignore_parsing (testsuite.test_parser.ParserTestCase) ... ok
test_vanilla_ignore_parsing (testsuite.test_parser.ParserTestCase) ... ok
test_check_diff (testsuite.test_shell.ShellTestCase) ... ok
test_check_noarg (testsuite.test_shell.ShellTestCase) ... ok
test_check_non_existent (testsuite.test_shell.ShellTestCase) ... ok
test_check_simple (testsuite.test_shell.ShellTestCase) ... ok
test_check_stdin (testsuite.test_shell.ShellTestCase) ... ok
test_print_usage (testsuite.test_shell.ShellTestCase) ... ok
test_normalize_paths (testsuite.test_util.UtilTestCase) ... ok

======================================================================
FAIL: test_checkers_testsuite (testsuite.test_all.PycodestyleTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/packages/tmp/pycodestyle-2.4.0/testsuite/test_all.py", line 39, in test_checkers_testsuite
    msg='%s failure(s)' % report.total_errors)
AssertionError: 7 is not false : 7 failure(s)

----------------------------------------------------------------------
Ran 53 tests in 0.698s

FAILED (failures=1)
Test failed: <unittest.runner.TextTestResult run=53 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=53 errors=0 failures=1>

--- a/pycodestyle.py
+++ b/pycodestyle.py
@@ -258,10 +258,10 @@ def trailing_blank_lines(physical_line, lines, line_number, total_lines):
     """
     if line_number == total_lines:
         stripped_last_line = physical_line.rstrip()
-        if not stripped_last_line:
+        if physical_line and not stripped_last_line:
             return 0, "W391 blank line at end of file"
         if stripped_last_line == physical_line:
-            return len(physical_line), "W292 no newline at end of file"
+            return len(lines[-1]), "W292 no newline at end of file"
 
 
 @register_check

